### ISSUE TITLE
⬆️ Allow chunked public upload

### ISSUE THREAD

**Issue by tessus** [nextcloud/server#4109]
State:
- [x] Internal DAV endpoints support chunking (ChunkingV2Plugin)
- [x] Public DAVv2 implemented allowing same logic as internal shares
- [x] Frontend supports chunking
- [ ] DAV support for Chunking on public shares
  - [ ] Requires [UploadHome](https://github.com/nextcloud/server/blob/92ba8fdcb5b0f6a8319c2b3050d54b5184d282df/apps/dav/lib/Upload/UploadHome.php#L18) to also support shares instead of only user sessions

---

**Original issue:**

I've noticed that for the file drop or any other screen that allows one to upload a file, there's no way to know in advance if the upload will fail.

Let's say the max upload size is 100 MB and someone tries to upload 105 MB. The upload will fail, but the person still uploaded 100MB for nothing. It would be nice, if there was a way to handle this scenario.
The easiest would be to show a message that tells an uploader the max upload size.

What do you think?

**Comment by MorrisJobke** [comment]
We plan to also do chunking on the web UI upload which would mean that we don't have a limit anymore.

Sadly I can't find the issue for this - @rullzer @jancborchardt @ChristophWurst Could you find it or was there none, yet?

**Comment by tessus** [comment]
Sounds good, but I suspect that implementing this chunking feature will take a while (months, years).

In the meantime a message would be great and very helpful. Adding the message won't be much of an effort for someone who knows the code.

**Comment by lesykm** [comment]
May be this could be reused:
https://github.com/owncloud/core/pull/28415/files

**Comment by MorrisJobke** [comment]
Web UI does chunking since Nextcloud 13 - we ported exactly this pr in #7056. Don't know if files drop and unauthenticated is also supported as of now. cc @danxuliu

**Comment by aryasenna** [comment]
Any new update to this? Any chance this will come on next major release?

**Comment by jay-tuckey** [comment]
This feature would be very handy for the university I work for, as we would like to be able to use the "file drop (upload only)" feature for student work submission, including videos projects which can be very large, ~2GiB. Currently we are testing a workaround by just increasing the max upload size, but then the student has no progress visible on how their upload is going or whether it's working. Any progress on this feature for public uploads?

**Comment by ChristophWurst** [comment]
@jay-tuckey do you have an Enterprise support subscription? Just asking because then we might be able to work on this in a professional services project.

**Comment by jay-tuckey** [comment]
@christophWurst unfortunately not at the moment, we may get a subscription in the future if we end up using nextcloud for more areas.

**Comment by tessus** [comment]
3 years later... Basic core function are apparently not a priority.

**Comment by jospoortvliet** [comment]
@tessus what is basic core functionality is in the eye of the beholder. As both private AND enterprise user I've never needed this. There is an easy work-around (just create a guest account) so I'm glad the team focuses on more important things.

**Comment by X-Coder** [comment]
I just wasted some hours figuring why the upload works everywhere but not on public sharing until I have seen it has to do with missing chunked file upload.

Could you please at least add a hint in the official docs about non available chunk-uploading for public sharing if you don't plan to fix this?

**Comment by FlashyDuck** [comment]
This is also affects WebDAV connections.

When using Cloudflare proxy uploads larger than 100MB fail. Chunking apparently needs to be forced by the user, but how?

**Comment by djdarkstep** [comment]
Yes, this bug totally breaks public uploads when using Cloudflare as proxy. When uploading from a public upload-enabled link, the entire file is sent in one single PUT request: /public.php/webdav/filename.ext

The free Cloudflare plan limits HTTP POST request size to 100MB, so if you try to publicly upload a file that is larger than 100MB, the upload fails.

**Comment by Rick45** [comment]
I have an use case where i need to have the public share with upload, There is any workaround for this?

**Comment by Rick45** [comment]
I have a scenario where i need to upload files up to 5G from a shared folder,
as workaround i have added `client_max_body_size 5G;`  in nextcloud/config/nginx/site-confs file

**Comment by Lynxiayel** [comment]
> I have a scenario where i need to upload files up to 5G from a shared folder, as workaround i have added `client_max_body_size 5G;` in nextcloud/config/nginx/site-confs file

I've run into same problem, but cannot find `nextcloud/config/nginx/site-confs`, there's only `config.php` in the config folder, can I have some tips? thanks.

**Comment by nooblag** [comment]
Try looking somewhere like `/etc/nginx/sites-available/` if you're using nginx.

However, sadly, it doesn't seem to be a fix anyway. I tried `client_max_body_size 5G;` and `client_max_body_size 0;` and numerous other large and small values (restarting nginx after each change/each time) and still, nothing over 100MB uploads successfully.

**Comment by Nirostar** [comment]
> Try looking somewhere like `/etc/nginx/sites-available/` if you're using nginx.
> 
> However, sadly, it doesn't seem to be a fix anyway. I tried `client_max_body_size 5G;` and `client_max_body_size 0;` and numerous other large and small values (restarting nginx after each change/each time) and still, nothing over 100MB uploads successfully.

Do you use cloudflare or some other proxy? Cloudflare does limit the upload to 100MB :/

**Comment by alfonsrv** [comment]
Shouldn't use GitHub issues for 101 tech support.  

Enabling a bigger client body in nginx/apache2 definitely fixes the undesired behaviour for public links.

**Comment by nooblag** [comment]
Yes, using Cloudflare---hence the desire for chunking to be implemented to fix the problem, like it is for logged-in users.

**Comment by Nirostar** [comment]
> Shouldn't use GitHub issues for 101 tech support.
> 
> Enabling a bigger client body in nginx/apache2 definitely fixes the undesired behaviour for public links.

Yeah but there are a lot of scenarios where you can't influence the max client body (Proxies, hosted systems etc.) Implementing chunked uploads for anonymous users would solve this instantly.

**Comment by Slopz** [comment]
Anyone found any workarounds? I really don't want to get rid of Cloudflare...and I also don't want to create user logins for every share link I send...

**Comment by nooblag** [comment]
Sadly, no. I hope the developers add file chunking to public uploads (just as is available for logged in users) sometime soon. 🤞 We do not want to get rid of Cloudflare either.

**Comment by HerbertHorst** [comment]
Is there any update on this? Im on version "[Nextcloud Hub 4](26.0.2)" and ran into this problem last week. Took me a few days to find what the problem is, but didnt found a solution yet.

**Comment by ufo56** [comment]
Any news ?

**Comment by joshtrichards** [comment]
#30069 may open the door to this functionality from the looks of it

**Comment by austin3410** [comment]
Verified this is still an issue on 27.1.4.... at least 6 years later.

**Comment by joshtrichards** [comment]
Well, it's an enhancement idea not a bug, and this is an open source project. The issue is still open, which indicates no one has jumped in and completed (volunteered) to do the work. You're welcome make an attempt at implementing it!

**Comment by ShaunCurrier** [comment]
Is there a feature bounty program for Nextcloud?  I'd love to throw some money that way for this issue.

**Comment by austin3410** [comment]
> Well, it's an enhancement idea not a bug, and this is an open source project. The issue is still open, which indicates no one has jumped in and completed (volunteered) to do the work. You're welcome make an attempt at implementing it!

Without any sort of warning, or in-app acknowledgement that File Drop links are meant for tiny files and anything that may be a "large" file may not work, I would still consider it a bug or at the very least a poorly implemented feature.

I had to spend an entire day searching through logs to finally figure out that the upload methods for authenticated vs unauthenticated sessions are completely different. If there was at least some sort of warning explaining the handicaps of using File Drop links I would've reconsidered my approach instead of wasting time searching for a problem on my end that wasn't there. ¯\_(ツ)_/¯

**Comment by alfonsrv** [comment]
You got some free extra knowledge on how the web works through troubleshooting it though 💭

**Comment by austin3410** [comment]
> You got some free extra knowledge on how the web works through troubleshooting it though 💭

lol fair XD

**Comment by nooblag** [comment]
Great discussion around this.

I only have this to add:

> Unfortunately NC itself has no way of knowing the limit because it's environment dependent [...]

That part (at the server end where Nextcloud is running) is discoverable through `ini_get('upload_max_filesize')` at least...

Yes there would be a challenge detecting with anything much beyond that (i.e. proxies and so on) but that's our point---that all of this would be fixed with a chunked file upload. :)

**Comment by TimKraemer** [comment]
I control everything from metal to nextcloud, which would be debian -> docker -> nginx proxy -> php -> nextcloud but I still can't manage to allow guests without registration to upload huge files to file drop - did anyone succeed with a setup like mine?

**Comment by realies** [comment]
> I control everything from metal to nextcloud, which would be debian -> docker -> nginx proxy -> php -> nextcloud but I still can't manage to allow guests without registration to upload huge files to file drop - did anyone succeed with a setup like mine?

Same here, and not yet. Maybe in another six years.

**Comment by LZNetwork-Main** [comment]
Hello, any updates on this?

**Comment by EuleMitKeule** [comment]
+1 on this.

**Comment by jospoortvliet** [comment]
For those waiting for this - the public file view is now also nearly finished with its Vue migration, which will allow enabling chunked upload. I'm not sure it will make it in Hub 9 (September), or just miss the deadline and come in Hub 10 (early next year), but it seems very close. Personally also looking forward to it, the new vue-ified public link shares will let us focus on some other nice sharing related features.

**Comment by realies** [comment]
omfg 7 years and it's finally complete 🍾

**Comment by Slopz** [comment]
What a time to be alive.

**Comment by szaimen** [comment]
🎉

**Comment by ivanjx** [comment]
is this not shipped with hub 10 yet?

![Image](https://github.com/user-attachments/assets/4c316eab-b320-4aac-ba76-d7b0c22c2081)

![Image](https://github.com/user-attachments/assets/875d5e5f-d97d-4fc4-8bbb-3d16c1c989fb)

![Image](https://github.com/user-attachments/assets/de952b75-d35e-4103-b52a-886c99812fbd)

**Comment by susnux** [comment]
This is sadly still not enabled. So the frontend would be ready for it but in the backend the DAV Plugin (ChunkingV2) is currently not working without a user session.
For anyone who wants to work on this: Have a look at `UploadHome` within the DAV app that needs to support public shares.

**Comment by sorbaugh** [comment]
Assigning @provokateurin

**Comment by aabrilfl13** [comment]
IS this going to be done?

**Comment by susnux** [comment]
> IS this going to be done?

Yes for Nextcloud 32
